src/components/Wallet/NavbarBalance.jsx

```diff
@@ -0,0 +1,147 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import styled from 'styled-components'
+
+export default class NavbarBalance extends Component {
+  static displayName = 'NavbarBalance'
+
+  static propTypes = {
+    balance: PropTypes.string,
+    currency: PropTypes.string,
+    network: PropTypes.string,
+    testButton: PropTypes.bool
+  }
+
+  static defaultProps = {
+    currency: 'ETHER*',
+    balance: '0.00',
+    network: 'rinkeby'
+  }
+
+  constructor(props) {
+    super(props)
+    this.state = {
+      balance: '0.00'
+    }
+  }
+
+  componentDidMount() {
+    this.mounted = true
+    let { balance: bal1 } = this.state
+    let { balance: bal2 } = this.props
+    bal1 = parseFloat(bal1)
+    bal2 = parseFloat(bal2)
+    if (bal1 !== bal2) {
+      this.changeBalance(bal1, bal2)
+    }
+  }
+
+  componentWillUnmount() {
```

Comment:
nit: linter requires lifecycle functions to be placed in order that they occur, so you'll need to move this function below `componentDidUpdate`.